integration-tests/ahm/src/mock.rs

```diff
@@ -113,23 +110,21 @@ pub fn enqueue_dmp(msgs: Vec<InboundDownwardMessage>) {
 // both the DMP messages sent from the relay chain to asset hub, which will be used to perform the
 // migration, and the relay chain payload, which will be used to check the correctness of the
 // migration process.
-pub fn rc_migrate<RcMigrator: RcPalletMigrationChecks>(
-	mut relay_chain: RemoteExternalities<PolkadotBlock>,
-) -> (Vec<InboundDownwardMessage>, RcMigrator::RcPayload) {
+pub fn rc_migrate(
```

Comment:
should we drop these abstractions? the `rc_migrate` and `ah_migrate` and have these code just within the `pallet_migration_works` function?

Reply:
I think its fine now, the main testing function looks quite clean. We can switch these functions to accept tokio channels in the if we want to run it in parallel.